inc/SalesData/WooToNative/Customers/Customers.php

```diff
@@ -23,38 +25,261 @@
+		return $this->get_woocommerce_customers( $limit, $offset );
+	}
+
+	/**
+	 * Total customer count
+	 *
+	 * @since 2.4.0
+	 *
+	 * @return int
+	 */
+	public function get_total_items_count(): int {
+		$this->total_customer_count = $this->get_woocommerce_customers_count();
+		return $this->total_customer_count;
+	}
+
 	/**
 	 * Extract customer data
 	 *
 	 * @since 2.4.0
 	 *
-	 * @param int|object $customer Customer id or object.
+	 * @param object $customer customer order object.
 	 *
-	 * @return mixed
+	 * @return self
+	 * @throws \Throwable Return throws.
 	 */
-	public function extract( $customer ) {
-		return $this->customer;
+	public function extract( $customer = null ): self {
+		try {
+			$this->current_customer = (object) $customer;
+			return $this;
+		} catch ( \Throwable $th ) {
+			throw $th;
+		}
 	}
 
 	/**
-	 * Transform the customer data to native customer
+	 * Get WooCommerce customers from orders
 	 *
 	 * @since 2.4.0
 	 *
+	 * @param int $limit  Number of items to fetch from source.
+	 * @param int $offset Number of items to skip from source.
+	 *
 	 * @return array
+	 * @throws \Throwable Return throws.
+	 */
+	public function get_woocommerce_customers( $limit = 5, $offset = 1 ) {
+		try {
+			$args      = array(
+				'status'     => array( 'completed', 'processing', 'on-hold' ),
+				'limit'      => $limit,
+				'paged'      => $offset,
+				'return'     => 'ids',
+				'meta_query' => array(
+					array(
+						'key'     => '_is_tutor_order_for_course',
+						'compare' => 'EXISTS',
+					),
+				),
+			);
+			$orders    = wc_get_orders( $args );
+			$customers = array();
+			$user_map  = array();
+
+			foreach ( $orders as $order_id ) {
+				$order   = wc_get_order( $order_id );
+				$user_id = $order->get_user_id();
+				if ( ! in_array( $user_id, $user_map, true ) ) {
+					$customers[] = array(
+						'id'    => $user_id,
+						'name'  => $order->get_billing_first_name() . ' ' . $order->get_billing_last_name(),
+						'email' => $order->get_billing_email(),
+					);
+					$user_map[]  = $user_id;
+				}
+			}
+
+			return $customers;
+		} catch ( \Throwable $th ) {
+			throw $th;
+		}
+	}
+
+	/**
+	 * Get total WooCommerce customers count from orders
+	 *
+	 * @since 2.4.0
+	 *
+	 * @return int
+	 * @throws \Throwable Return throws.
 	 */
-	public function transform(): array {
-		return array();
+	public function get_woocommerce_customers_count() {
+		global $wpdb;
+		try {
+			$is_hpos_enabled = false;
+
+			if ( class_exists( \Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController::class ) ) {
+				$controller      = wc_get_container()->get( \Automattic\WooCommerce\Internal\DataStores\Orders\CustomOrdersTableController::class );
+				$is_hpos_enabled = $controller->custom_orders_table_usage_is_enabled();
+			}
+
+			if ( $is_hpos_enabled ) {
+				// HPOS query.
+				$query = "
+				SELECT DISTINCT c.customer_id, c.user_id, c.email, c.first_name, c.last_name
```

Comment:
Since we just need the count value we can write query like the following:
`SELECT COUNT(*)`
Instead of this 
`SELECT DISTINCT c.customer_id, c.user_id, c.email, c.first_name, c.last_name`